src/h5web/providers/hsds/models.ts

```diff
@@ -11,6 +9,72 @@ import type {
 } from '../hdf5-models';
 import type { Entity } from '../models';
 
+/* --------------------- */
+/* ----- RESPONSES ----- */
+
+export interface HsdsRootResponse {
+  root: HDF5Id;
+}
+
+export interface HsdsGroupResponse {
+  id: HDF5Id;
+  linkCount: number;
+  attributeCount: number;
+}
+
+export interface HsdsDatasetResponse {
+  id: HDF5Id;
+  shape: HsdsShape;
+  type: HsdsType;
+  attributeCount: number;
+}
+
+export interface HsdsDatatypeResponse {
+  id: HDF5Id;
+  type: HsdsType;
+}
+
+export interface HsdsAttributesResponse {
+  attributes: Omit<HsdsAttributeWithValueResponse, 'value'>[];
+}
+
+export interface HsdsAttributeWithValueResponse {
+  name: string;
+  shape: HsdsShape;
+  type: HsdsType;
+  value: HDF5Value;
+}
+
+export interface HsdsLinksResponse {
+  links: HsdsLink[];
+}
+
+export interface HsdsValueResponse {
+  value: HDF5Value;
+}
+
+/* ---------------------------- */
+/* ----- ENTITIES & VALUES----- */
+
+export type HsdsEntity<T extends Entity = Entity> = T & { id: HDF5Id };
+
+export type HsdsLink = HDF5HardLink | HDF5SoftLink | HsdsExternalLink;
+
+export interface HsdsExternalLink extends Omit<HDF5ExternalLink, 'file'> {
+  h5domain: string;
+}
+
+export type HsdsComplex = HsdsComplex[] | HsdsComplexValue;
+export type HsdsComplexValue = [number, number];
+
+/* ------------------------ */
+/* ----- SHAPE & TYPE ----- */
+
+export interface HsdsShape {
+  class: 'H5S_SIMPLE' | 'H5S_SCALAR' | 'H5S_NUL';
+  dims?: HDF5Dims;
+}
```

Comment:
The new HSDS shape type. I don't specify that `dims` is only provided with `H5S_SIMPLE`, which allows the conversion function to be much simpler. 😄 We don't need separate shape types for inferring things later on, so it's not worth having them. I might do something similar with `HSDSType`, I have looked yet.